src/Tizen.Applications.WidgetControl/Tizen.Applications/WidgetControl.cs

```diff
@@ -314,6 +314,74 @@ public static string[] GetWidgetIds(string pkgId)
             return list.ToArray();
         }
 
+        /// <summary>
+        /// Gets main appid of the widget.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/widget.viewer</privilege>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown in case of failed conditions.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the API is not supported in this device.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when an application does not have the privilege to access this method.</exception>
+        public static string GetWidgetMainAppId(string widgetId)
```

Comment:
How about add MainAppId property on WidgetControl instead of this method?
``` c#
class WidgetControl {
     public string MainAppId { 
       get {
           return Interop.WidgetService.GetWidgetMainAppId(Id);
       }
     }
}
```